### PR TITLE
added symptom support, support lists subtypes for diagnosis and sympt…

### PR DESCRIPTION
**Pull request by kchapple**
@sjpadgett here's code... for some reason the last two changes you made to the import for issue_encounter and adding the subtype definitions haven't worked on my system (the don't come through to my database), but I think this should work.

### CODE REVIEW

**Comment by sjpadgett** [comment]
that's concerning. I don't see why.

**Comment by sjpadgett** [comment]
Yeah, good job @kchapple !
image

**Comment by kchapple** [comment]
Subtypes worked after I pulled your last commit. All good there! Symptom template is showing up. Still not getting issue_encounter on import, so my encounter templates don't have the observation section with the code. Not sure if it's me, or what

**Comment by sjpadgett** [comment]
I rechecked my commits and everything is there for issue_encounter. I hope there's not some weird mac vs windows thing going on.

**Comment by sjpadgett** [comment]
@kchapple Looks like we are going to have to track end dates for encounters! When does this end!!!
`<effectiveTime><low value='20181104080000'/><high nullFlavor='UNK'/></effectiveTime>`

To meet calculator 
image

**Comment by kchapple** [comment]
The encounter export service is half-implemented to use the events table to add the encounter duration to the start time to get the end. if that would work?

**Comment by sjpadgett** [comment]
yep that'd be the best option. Go ahead and turn off hard coded time and look at the event. I'm to tired tonight to fix in the import but will first thing tomorrow.

**Comment by kchapple** [comment]
@sjpadgett I was getting device orders from orders table with a type 'device' but changed to retrieve from care plans

**Comment by sjpadgett** [comment]
It can come from both! one is a procedure the other is a planned order. so two device types `device` for procedure and `device_order' for careplan.

**Comment by kchapple** [comment]
I think I see. I needed to add service to spit out the 'device recommended' models. reverted device order (which points to procedures)

**Comment by sjpadgett** [comment]
Do We still have device recommended

**Comment by kchapple** [comment]
Now we have both DeviceOrderService (procedure) and DeviceRecommendedService (care plan)

We'll probably need 'encounter recommended' and 'assessment recommended' in care plans also, but haven't run across them yet. Those look like the last two care plans that may be unimplimented.

**Comment by sjpadgett** [comment]
device order is a oid for qrda only where device as a procedure has ccda components so for a care plan order it is an activity where the procedure is the act on implementing the device. so the planned act for the device is this template.
```
<entry>
              <act classCode="ACT" moodCode="RQO" >
                <templateId root="2.16.840.1.113883.10.20.24.3.130" extension="2019-12-01"/>
                <code code="SPLY" codeSystem="2.16.840.1.113883.5.6" displayName="Supply" codeSystemName="ActClass"/>
                <entryRelationship typeCode="SUBJ">
                  <supply classCode="SPLY" moodCode="RQO">
                    <!-- Plan of Care Activity Supply -->
                    <templateId root="2.16.840.1.113883.10.20.22.4.43" extension="2014-06-09"/>
                    <!-- Device, Order -->
                    <templateId root="2.16.840.1.113883.10.20.24.3.9" extension="2019-12-01"/>
                    <id root="1.3.6.1.4.1.115" extension="6257ac0363623902e9d97ff2"/>
                    <text>Frailty Device</text>
                    <statusCode code="active"/>
                      <!-- QDM Attribute: Author dateTime -->
                      <author>
                        <templateId root="2.16.840.1.113883.10.20.24.3.155" extension="2019-12-01"/>
                        <time value='20200624081500'/>
                        <assignedAuthor>
                              <id nullFlavor="NA"/>
                        </assignedAuthor>
                      </author>                    <participant typeCode="DEV">
                      <participantRole classCode="MANU">
                        <playingDevice classCode="DEV">
                          <!-- QDM Attribute: Code -->
                            <code code="360006004" codeSystem="2.16.840.1.113883.6.96" codeSystemName="SNOMEDCT"/>
                        </playingDevice>
                      </participantRole>
                    </participant>
                  </supply>
                </entryRelationship>
              </act>
            </entry>
```

I know, it's nuts!

**Comment by kchapple** [comment]
OK, I think I got it. I changed the DeviceAppliedService to spit out models based on care plans. DeviceRecommendedService will do the same thing, but I'm not sure what the care_plan_type string for 'device recommended' care plan should be, so I just left as 'device' for now (see AbstractCarePlanService for list of types)

**Comment by sjpadgett** [comment]
Nope you're giving the procedure device I need for care plan device_order
src/Services/Qrda/qrda-export/catI-r5/qrda_templates/device_order.mustache

**Comment by kchapple** [comment]
ok, if this isn't right just send me a xml example and I'll straighten it out in the morning.

**Comment by kchapple** [comment]
@sjpadgett I forgot to check in the service files on my last commit last night. I also added a couple edits to medication export. 

A couple observations on meds (prescriptions table) 

1. start_date and end_date are not populated  on import. Currently using date_added, which is  a totally wrong date (is date that the DB row was created). sjp fixed this
2. they are date types rather than datetimes. re: sjpdon't know what you mean

**Comment by sjpadgett** [comment]
@kchapple not returning the template for me. does it for you?

**Comment by kchapple** [comment]
@sjpadgett For Dianne Newton: On  the left is our export. On the Right is the original.
Screen Shot 2022-04-17 at 12 23 05 PM

**Comment by kchapple** [comment]
@sjpadgett give me a min, something is broken, I'm getting old data (it was working so that's why I have the valid comparison.)

**Comment by kchapple** [comment]
Never mind my last comment about old data. I just came back to computer, and ran and thought it was broken because my session had expired. I logged in again, and got the same result as screenshot above. Looks like it's working to me.

**Comment by kchapple** [comment]
@sjpadgett I ran using the CareCoordination export rather than cqm tools, and got same result as in screenshot above. I had to fix the mkdir() function params to get it to complete download (see last commit).

File: src/Cqm/QrdaControllers/QrdaReportController.php [hunk 1]
```diff
@@ -100,10 +100,9 @@ public function downloadQrdaIAsZip($pids, $measures = '', $type = 'xml', $option
         $directory = $GLOBALS['OE_SITE_DIR'] . '/documents/' . 'cat1_reports';
         $directory .= ($bypid ? '/all_measures' : "/measures");
         if (!is_dir($directory)) {
-            if (!mkdir($directory, true, true) && !is_dir($directory)) {
+            if (!mkdir($directory, 0755, true) && !is_dir($directory)) {
```
Comment (sjpadgett) [inline src/Cqm/QrdaControllers/QrdaReportController.php:103]: silly mistakes! windows didn't care so doesn't complain thanks

**Comment by sjpadgett** [comment]
okay, maybe i'm missing something from your repo.

**Comment by sjpadgett** [comment]
Still doesn't work for me so I must be missing something.

**Comment by kchapple** [comment]
I think I'm up-to-date
```
{22-04-17 13:18}iMac:~/Dev/openemr/master@qrda-dates % git pull jerry qrda-dates
From https://github.com/sjpadgett/openemr
 * branch                qrda-dates -> FETCH_HEAD
Already up to date.
{22-04-17 13:18}iMac:~/Dev/openemr/master@qrda-dates % git status
On branch qrda-dates
Your branch is ahead of 'jerry/qrda-dates' by 9 commits.
  (use "git push" to publish your local commits)

nothing to commit, working tree clean
```
Here's my git log:
```
252aec0d9 (HEAD -> qrda-dates, origin/qrda-dates) fix for creating direcotry when exporting one patient QRDA
ad285402c Forgot to checkin the service files, some fixes in medicaiton export
f67245e21 change device-ordered to care plan, device applied to procedure
2a1fad2ab change device applied to use care plans table
9efba2497 add device recommended, revert device order to point to procedures
1715ad69b change device order service to fetch from care plan
ecd7e89ac Merge branch 'qrda-dates' of https://github.com/sjpadgett/openemr into qrda-dates
61df16c93 (jerry/qrda-dates) add careplan device order import
f8b981d13 add date_end to import parse add date_end to qdm encounter service
ec337d006 add encounter date end to database
ca3a8665f Merge branch 'qrda-dates' of https://github.com/sjpadgett/openemr into qrda-dates
6e2cc0c20 added symptom support, support lists subtypes for diagnosis and symptom, removed unused condition service class
af95acf05 missed subtype for problems
94aacd685 add subtype to classify problems encounter_dianosis, diagnosis and concern
963eb2e11 add classification to problems in list diagnosis or concern new concern problem template parse.
ccd87d24c add missing oid codetype
9252c5cf3 fix code_and_codesystem helper to resolve missing oid from valueset mod code lookup to include just a code lookup
692e430fe somehow these two scripts were missed.
f998d98db Merge branch 'master' of https://github.com/openemr/openemr into qrda-dates
ddc1d717d Qrda encouter issue (#5165)
249c1ceb8 (upstream/master, master) void checkout or reopen global and logic to support (#5156)
```

**Comment by kchapple** [comment]
You might have to `git pull mi-squared qrda-dates` and replace `mi-squared` with whatever my remote is called in your system.

**Comment by kchapple** [comment]
Or you could just merge this PR into your qrda-dates branch

**Comment by sjpadgett** [comment]
@kchapple Yeah I had to delete my local and repull your branch and finally works, thanks.
I also fixed begin and end dates in medication and will push to my PR after I fix new db conflicts.
I'll run new test on cms125 and should be green. I hope we haven't broken any previously tested templates.

**Comment by sjpadgett** [comment]
@kchapple Look at what we did!

image

**Comment by sjpadgett** [comment]
@kchapple merge or leave open. your choice

**Comment by kchapple** [comment]
Merge it!

**Comment by sjpadgett** [comment]
thank you Ken....

**Pull request merged**